include/graphblas/reference/blas3.hpp

```diff
@@ -918,6 +937,301 @@ namespace grb {
 
 	namespace internal {
 
+		template<
+			Descriptor descr,
+			class Monoid,
+			typename InputType, typename IOType,
+			typename RIT, typename CIT, typename NIT
+		>
+		RC fold_unmasked_generic(
+			IOType &x,
+			const Matrix< InputType, reference, RIT, CIT, NIT > &A,
+			const Monoid &monoid
+		) {
+			_DEBUG_PRINT( "In grb::internal::foldr_unmasked_generic( reference )\n" );
+			RC rc = SUCCESS;
+
+			if( grb::nnz( A ) == 0 ) {
+				_DEBUG_PRINT( "The input matrix is empty, nothing to compute\n" );
+				return rc;
+			}
+
+			if ( descr & descriptors::force_row_major && descr & descriptors::transpose_left ) {
+				_DEBUG_PRINT( "Masked fold with force_row_major and transpose_left is not supported\n" );
+				return RC::ILLEGAL;
+			}
+			if ( descr & descriptors::force_row_major && descr & descriptors::transpose_matrix ) {
+				_DEBUG_PRINT( "Masked fold with force_row_major and transpose_matrix is not supported\n" );
+				return RC::ILLEGAL;
+			}
+
+			const auto &A_raw = (descr & grb::descriptors::transpose_matrix || descr & grb::descriptors::transpose_left ) ?
+				internal::getCCS( A ) : internal::getCRS( A );
+			const size_t A_nnz = nnz( A );
+
+			const auto& op = monoid.getOperator();
+			RC local_rc = rc;
+			auto local_x = monoid.template getIdentity< typename Monoid::D3 >();
+
+#ifdef _H_GRB_REFERENCE_OMP_BLAS3
+	#pragma omp parallel default(none) shared(A_raw, x, rc, std::cout) firstprivate(local_x, local_rc, A_nnz, op)
```

Comment:
tabs here can be aligned with the `{`. Also, line length. The code so far typically does not use `default(none)` for OMP regions -- is it totally necessary here?